model_compression_toolkit/quantizers_infrastructure/keras/inferable_quantizers/weights_inferable_quantizers/weights_uniform_inferable_quantizer.py

```diff
@@ -50,31 +52,44 @@ def __init__(self,
                 max_range: max quantization range for quantizing weights
                 per_channel: whether to use per-channel quantization
                 channel_axis: axis along which to apply per-channel quantization
+                input_rank: number of dimensions of input tensor the quantizer quantizes
             """
             super(WeightsUniformInferableQuantizer, self).__init__(num_bits,
                                                                    min_range,
                                                                    max_range)
 
+            if per_channel:
+                assert input_rank is not None, f'Input rank is missing in per channel quantization'
+                assert channel_axis is not None, f'Channel axis is missing in per channel quantization'
+                assert len(
+                    self.min_range) >= 1, f'In per-channel quantization min/max numpy arrays should be of length >= 1 but is ' \
```

Comment:
just min not min/max (BTW, why do we need to enforce this? is it going to fail if the size is 1, hypothetically if there is 1 channel?)

Reply:
That's because in his super init there is a check that min and max are at the same length, but maybe it's better to split it up here for clarity.
And for the question - yeah. Do you see any issue with that?